### ISSUE TITLE
dired-do-query-replace-regexp problems

### ISSUE THREAD

**Issue by LefterisJP** [syohex/emacs-anzu#46]
Hello,

First of all thanks for making this awesome package! I have ran into an issue when using anzu that must be either some misconfiguration on my part or some clash between anzu and `dired-do-query-replace-regexp`.

Steps to reproduce:
1. Use `anzu-query-replace` on a buffer. It works fine.
2. Then do `find-name-dired` in that same directory (or possibly any other directory I guess), for all files with `*` wildcard and select all files with `t`.
3. One you press `Q` in the dired buffer which should execute `dired-do-query-replace-regexp` you get an error.

The error is:
```
query-replace-descr: Wrong type argument: listp, "_active"
```
where "_active" is basically the previous string you tried to replace.

Here is a backtrace. I was simply experimenting with the search string being the same as the replace string here. It happens also when they are different:
``` emacs-lisp
Debugger entered--Lisp error: (wrong-type-argument listp "_active")
  mapconcat(isearch-text-char-description ("_active" . "_active") "")
  query-replace-descr(("_active" . "_active"))
  query-replace-read-from("Query replace regexp in marked files" t)
  query-replace-read-args("Query replace regexp in marked files" t t)
  (let ((common (query-replace-read-args "Query replace regexp in marked files" t t))) (list (nth 0 common) (nth 1 common) (nth 2 common)))
  call-interactively(dired-do-query-replace-regexp nil nil)
  command-execute(dired-do-query-replace-regexp)
```

Any help would be greatly appreciated. I really like Anzu but I also want to have dired's search and replace functionality.

**Comment by syohex** [comment]
Thanks for reporting.

The difference variable type of `query-replace-defaults` between Emacs 24(dotted-pair) and
Emacs 25(list of dotted-pair) makes this error. I missed this error because I always use
development version.

I fixed this error at #47. Please check latest version and close this issue
if you resolve this issue.

**Comment by LefterisJP** [comment]
Thank you for your quick response. After your commit I no longer get the error when using dired. Seems fixed.
